app/controllers/updates_controller.rb

```diff
@@ -0,0 +1,26 @@
+class UpdatesController < ApplicationController
+  def show
+    update_paths = Dir.glob(Rails.root.join('app/views/updates/update_files/*').to_s)
+    @updates = update_file_paths_to_hash(update_paths).sort.reverse.to_h
+  end
+
+  def update_file_paths_to_hash(update_paths)
```

Comment:
There is too much logic here in the controller, controllers are only responsible for checking params and invoking models/services for preparing the view. We can pair together to refactor it if you want.